SConstruct

```diff
@@ -337,6 +337,11 @@ def scons():  # pylint: disable=too-many-locals,too-many-branches
             if value is not None:
                 env['ENV'][key] = value
 
+    config = Configure(env)
+    if not config.CheckHeader('stdatomic.h'):
+        Exit('stdatomic.h is required to compile DAOS, update your compiler or distro version')
```

Comment:
@jolivier23 By moving this check from compiler_setup.py to here we can ensure that the build will fail before attempting to build the dependencies.  As this PR is disallowing the use of CentOS 7 then failing early and correctly is important.

Reply:
My general preference is to do this using prereq.  We used to have a stdatomic component defined in site_scons/components.  But I have no major objection here.

Reply:
The reasoning is that in the past, there have been some bugs in configure code such that I prefer keeping configure tests in a single pass, if possible.

Reply:
We've got multiple configure passes already though, I've never heard that requirement before.

Most of the prereqs checks are for installing separate components (i.e. you can resolve failures by installing extra packages) however this one is only resolvable by upgrading away from EL7 so I'm OK with with one being different/earlier.

The advantage to doing it in prereqs however would be that we'd check with $COMPILER rather than gcc which might make a difference although I don't think it would in practice.

Reply:
@ashleypittman it should  go after line 359 so that things like ~/.scons_localrc are read which may change the compiler path used.